### PR TITLE
chore: apply some clippy lints

### PR DESCRIPTION
**Pull request by nyurik**
@rustbot label +stable-nominated

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -147,19 +147,38 @@ members = [
...
+cast_sign_loss = "allow"
+expl_impl_clone_on_copy = "allow"
+must_use_candidate = "allow"
+pub_underscore_fields = "allow"
+struct_field_names = "allow"
+uninlined_format_args = "allow"
+unreadable_literal = "allow"
+used_underscore_binding = "allow"
+used_underscore_items = "allow"
+verbose_bit_mask = "allow"
```
Comment (tgross35) [inline Cargo.toml:184]: I think we could simplify this a lot, a lot of the lints aren't enabled by default:

```suggestion
# We are okay with the current state of these lints
explicit_iter_loop = "warn"
manual_assert = "warn"
map_unwrap_or = "warn"
non_minimal_cfg = "allow"  # for some reason cfg_if! sometimes trigger this
ptr_as_ptr = "warn"
unnecessary_semicolon = "warn"

# FIXME(clippy): these should be fixed if possible
expl_impl_clone_on_copy = "allow"
uninlined_format_args = "allow"
used_underscore_binding = "allow"
```

**Review by tgross35 — CHANGES_REQUESTED** [review]

**Review by tgross35 — APPROVED** [review]
Looks great, thank you for working on these! As with the others, needs a squash
